public/main/client/handlers/no-core.js

```diff
@@ -22,14 +22,18 @@ const persistState = data => storage.persistState(data).then(() => true)
 function changePassword ({ oldPassword, newPassword }) {
   return validatePassword(oldPassword)
     .then(function (isValid) {
-      if (isValid) {
-        wallet.getWallets().forEach(function (walletId) {
-          const seed = wallet.getSeed(walletId, oldPassword)
-          auth.setPassword(newPassword)
-          wallet.setSeed(seed, newPassword)
-        })
+      if (!isValid) {
+        return isValid
       }
-      return isValid
+      return auth
+        .setPassword(newPassword)
+        .then(function () {
+          wallet.getWallets().map(function (walletId) {
+            const seed = wallet.getSeed(walletId, oldPassword)
+            wallet.setSeed(seed, newPassword)
+          })
+        })
+        .then(() => isValid)
```

Comment:
this update is because `setPassword` is async, and not related to the wallet, so doing it once is enough. I found this problem while trying to make the async settings API work

Reply:
What about something even simpler?

```js
      return isValid && auth // <<<< instead of the `if` above
        .setPassword(newPassword)
        .then(function () {
          wallet.getWallets().forEach(function (walletId) { // <<<< use `forEach` as we are not mapping anything
            const seed = wallet.getSeed(walletId, oldPassword)
            wallet.setSeed(seed, newPassword)
          })
          return true // <<<< `isValid` is `true` if we reached here
        })
```

Reply:
Applied here https://github.com/autonomoussoftware/metronome-wallet-desktop/pull/484/commits/8ca46d4f535046a5eaeeaf2594a7d10893ae2585

I did not notice I was using `map`, thanks for pointing out!. I switched the `return true` - that reads better. I did keep the `if clause` because I'm not a huge fan of how it reads mixing the flag with a whole function call there inlined - I think this makes it easier to read.